backend/src/main/java/codezap/template/service/facade/TemplateApplicationService.java

```diff
@@ -28,63 +29,70 @@
 @Service
 @RequiredArgsConstructor
 public class TemplateApplicationService {
-    private final TemplateTagService templateTagService;
+
     private final TemplateService templateService;
-    private final ThumbnailService thumbnailService;
     private final SourceCodeService sourceCodeService;
+    private final MemberService memberService;
+    private final CategoryService categoryService;
+    private final TagService tagService;
+    private final ThumbnailService thumbnailService;
 
     @Transactional
-    public Long createTemplate(Member member, Category category, CreateTemplateRequest createTemplateRequest) {
+    public Long createTemplate(Member member, CreateTemplateRequest createTemplateRequest) {
+        Category category = categoryService.fetchById(createTemplateRequest.categoryId());
+        category.validateAuthorization(member);
```

Comment:
이게 이 pr 내용에 포함되어도 되는 리뷰인지는 모르겠지만 아니라도 기록은 있어야 할 것 같아 남겨요!
(제우스 판단 하에 반영 여부를 결정하시면 좋을 것 같슴다)

```suggestion
        Category category = categoryService.findOwnCategoryById(member, createTemplateRequest.categoryId()); // 메서드 이름은 더 적절한 것으로 바꾸기
```

이 로직이 2번 사용되고 있는데 두번 다 권한 확인까지 하고 있어요!
categoryService단에서 권한까지 확인한 후 권한이 있는 경우에만 카테고리를 반환하는 로직으로 가도 좋을듯합니다!